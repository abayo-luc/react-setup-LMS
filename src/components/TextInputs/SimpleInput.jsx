import React from 'react';
import PropTypes from 'prop-types';
import './input.scss';

const SimpleInput = (props) => {
  const {
    type, name, value, id, onChange, placeholder, label,
  } = props;
  return (
    <label htmlFor="inputEmail4">
      {label}

      <input
        type={type}
        className="form-control"
        name={name}
        id={id}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </label>
  );
};
SimpleInput.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  id: PropTypes.string,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
SimpleInput.defaultProps = {
  id: '',
};
export default SimpleInput;
