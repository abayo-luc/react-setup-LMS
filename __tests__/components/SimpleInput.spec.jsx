import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SimpleInput from '../../src/components/TextInputs/SimpleInput';

const props = {
  type: 'text',
  label: 'Something',
  name: 'something',
  value: 'hello world',
  placeholder: 'Something',
  onChange: jest.fn(),
};
const warper = shallow(<SimpleInput {...props} />);
describe('Simple Input', () => {
  test('should match the snapshot', () => {
    expect(toJson(warper)).toMatchSnapshot();
  });
});
