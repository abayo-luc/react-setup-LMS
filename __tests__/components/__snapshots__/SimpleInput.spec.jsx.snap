// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Input should match the snapshot 1`] = `
<label
  htmlFor="inputEmail4"
>
  Something
  <input
    className="form-control"
    id=""
    name="something"
    onChange={[MockFunction]}
    placeholder="Something"
    type="text"
    value="hello world"
  />
</label>
`;
